plugins/c8y_log_plugin/src/main.rs

```diff
@@ -154,6 +126,72 @@ async fn run(
     }
 }
 
+pub async fn process_mqtt_messages(
+    message: Option<Message>,
```

Comment:
I would call this function only when there is *actually* a message.

```suggestion
pub async fn process_mqtt_messages(
    message: Message,
```

Reply:
fixed